client.go

```diff
@@ -485,68 +486,69 @@ func (c *Client) Do(req *Request, resp *Response) error {
 		return fmt.Errorf("unsupported protocol %q. http and https are supported", uri.Scheme())
 	}
 
+	c.mOnce.Do(func() {
+		c.m = make(map[string]*HostClient)
+		c.ms = make(map[string]*HostClient)
```

Comment:
```suggestion
		c.mLock.Lock()
		c.m = make(map[string]*HostClient)
		c.ms = make(map[string]*HostClient)
		c.mLock.Unlock()
```
I know it's very unlikely but lets add this just to be safe in case someone calls `.Do()` and `.CloseIdleConnections()` at exactly the same time.